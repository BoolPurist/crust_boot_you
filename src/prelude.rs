use std::io;

pub use crate::constants;
pub use anyhow::Context;

pub type IoResult<T = ()> = Result<T, io::Error>;
pub type AppError = anyhow::Error;
pub type AppResult<T = ()> = Result<T, AppError>;

pub type ReturnToUser = AppResult<String>;
pub type PathResult = AppResult<PathBuf>;
pub type AppIoResult<T = ()> = Result<T, AppIoError>;

pub use crate::app_traits::file_manipulator::FileManipulator;
pub use crate::app_traits::path_provider::PathProvider;
pub use crate::file_management::AppIoError;
pub use crate::NotEmptyText;
pub use std::path::{Path, PathBuf};

#[cfg(test)]
pub use mockall::automock;
