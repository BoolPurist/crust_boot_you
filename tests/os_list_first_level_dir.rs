mod common;
use common::prelude::*;

#[test]
#[named]
fn list_all_nodes_1_level_only_without_files() {
    let setup = TestSetup::new(actual_expected!());
    let file_mani = setup.os_mani();
    let tmp_path = setup.path_to_temp();
    let actual = file_mani.list_first_level_dir(&tmp_path).unwrap();
    let actual_stripped = common::strip_away_changing_temp_prefix(&tmp_path, actual.as_slice());
    insta::assert_debug_snapshot!(actual_stripped);

    setup.assert_with_expected();
}
#[test]
#[named]
fn list_nothing_for_there_no_folders() {
    let setup = TestSetup::new(actual_expected!());
    let file_mani = setup.os_mani();
    let tmp_path = setup.path_to_temp();
    let actual = file_mani.list_first_level_dir(&tmp_path).unwrap();
    let actual_stripped = common::strip_away_changing_temp_prefix(&tmp_path, actual.as_slice());
    insta::assert_debug_snapshot!(actual_stripped);

    setup.assert_with_expected();
}
