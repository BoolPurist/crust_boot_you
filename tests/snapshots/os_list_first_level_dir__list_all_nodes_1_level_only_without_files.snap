---
source: tests/os_list_first_level_dir.rs
expression: actual_stripped
---
[
    "bbb",
    "aaa",
]
